### PR TITLE
Clean up config environment variables for server and agent

### LINKED ISSUE #209: Phase out environment variables prefixed DRONE_, use WOODPECKER_
Currently (as of v0.13.0), every environment variable for configuring woodpecker seems to come in two flavours: One prefixed `DRONE_` and one prefixed `WOODPECKER_`.

The coexistence of these two versions of variables with identical meaning makes documentation, help text, and error messages hard to read (see my PR https://github.com/woodpecker-ci/woodpecker/pull/208 for an example).

To improve the user experience, I recommend to make a decision regarding when to drop the `DRONE_` spelling and communicate that decision clearly.

Since woodpecker is still in the v0.x.x major version realm, I would argue such a change could happen with any minor version bump.

### PR DESCRIPTION
**Pull request by marians**
close #209

The goal here is to make consistent use of configuration environment variables prefixed `WOODPECKER_`. Where several variants existed, this PR aims to remove all but one option, leaving the most explicit.

This PR only changes server and agent code, but not documentation, in order to keep the PR digestible. Once we have consensus that this is correct, I'll change docs accordingly.

User (rather: admin) facing changes in this PR:

- In general, support for all server and agent config environment variables (env vars) starting with `DRONE_` is removed. The according `WOODPECKER_*` variables must be used instead.
- The env var `WOODPECKER_HOST` replaces `DRONE_HOST`, and `DRONE_SERVER_HOST`.
- The env var `WOODPECKER_AGENT_SECRET` is used to configure the shared secret which agents use to authenticate against the server. It replaces `WOODPECKER_SECRET`, `DRONE_SECRET`, `WOODPECKER_PASSWORD`, `DRONE_PASSWORD`, and `DRONE_AGENT_SECRET`.
- The env var `WOODPECKER_DATABASE_DRIVER` replaces `DRONE_DATABASE_DRIVER` and `DATABASE_DRIVER`.
- The env var `WOODPECKER_DATABASE_DATASOURCE` replaces `DRONE_DATABASE_DATASOURCE` and `DATABASE_CONFIG`.

## Todo
- [x] cleanup cli flags
- [x] add migration notes to docs 
  - instead of `DRONE_` use `WOODPECKER_`
  - renamed env-vars and their new names (`WOODPECKER_AGENT_SECRET`, `WOODPECKER_SERVER_HOST`, `WOODPECKER_DATABASE_DRIVER`, `WOODPECKER_DATABASE_DATASOURCE`)
- [x] update docs (docker-compose samples, server config)
- [x] update sample docker-compose

## Wait for
- [x] #356

### CODE REVIEW

**Comment by marians** [comment]
TODO: As pointed out in https://github.com/woodpecker-ci/woodpecker/issues/209#issuecomment-855300074 the XDG_ variable should not be touched here.

**Comment by marians** [comment]
I just reverted the removal of XDG_CACHE_HOME.

**Comment by marians** [comment]
@laszlocph This is ready for a review.

**Comment by 6543** [comment]
got conflicts :O

**Comment by marians** [comment]
FYI I won't be able to work on this PR in the foreseeable future. Feel free to take over, modify, and merge when ready.

**Comment by 6543** [comment]
I'll review it soon :)

File: cli/exec/exec.go [hunk 1]
```diff
@@ -86,207 +86,207 @@ var Command = cli.Command{
...
+			EnvVar: "WOODPECKER_PREV_COMMIT_REF",
 			Name:   "prev-commit-ref",
-			EnvVar: "DRONE_PREV_COMMIT_REF,WOODPECKER_PREV_COMMIT_REF",
 		},
 		cli.StringFlag{
+			EnvVar: "WOODPECKER_PREV_COMMIT_REFSPEC",
 			Name:   "prev-commit-refspec",
-			EnvVar: "DRONE_PREV_COMMIT_REFSPEC,WOODPECKER_PREV_COMMIT_REFSPEC",
 		},
 		cli.StringFlag{
```
Comment (6543) [inline cli/exec/exec.go:255]: own flag.go file?

**Review by 6543 — APPROVED** [review]

File: cli/exec/exec.go (hunk 1 above)
Comment (anbraten) [reply cli/exec/exec.go:255]: Would be nice and would match how it has been done for server and agent.

File: cmd/agent/flags.go [hunk 2]
```diff
@@ -22,83 +22,83 @@ import (
 
 var flags = []cli.Flag{
 	cli.StringFlag{
-		EnvVar: "DRONE_SERVER,WOODPECKER_SERVER",
+		EnvVar: "WOODPECKER_SERVER",
 		Name:   "server",
 		Usage:  "drone server address",
```
Comment (anbraten) [inline cmd/agent/flags.go:27]:
```suggestion
		Usage:  "server address",
```

File: cmd/agent/flags.go [hunk 3]
```diff
@@ -22,83 +22,83 @@ import (
...
 		Value:  "localhost:9000",
 	},
 	cli.StringFlag{
-		EnvVar: "DRONE_USERNAME,WOODPECKER_USERNAME",
+		EnvVar: "WOODPECKER_USERNAME",
 		Name:   "username",
 		Usage:  "drone auth username",
```
Comment (anbraten) [inline cmd/agent/flags.go:33]:
```suggestion
		Usage:  "auth username",
```

File: cmd/agent/flags.go [hunk 4]
```diff
@@ -22,83 +22,83 @@ import (
...
 	},
 	cli.BoolTFlag{
-		EnvVar: "DRONE_DEBUG_NOCOLOR,WOODPECKER_DEBUG_NOCOLOR",
+		EnvVar: "WOODPECKER_DEBUG_NOCOLOR",
 		Name:   "nocolor",
 		Usage:  "disable colored debug output",
 	},
 	cli.StringFlag{
-		EnvVar: "DRONE_HOSTNAME,HOSTNAME,WOODPECKER_HOSTNAME,HOSTNAME",
+		EnvVar: "WOODPECKER_HOSTNAME,HOSTNAME",
```
Comment (anbraten) [inline cmd/agent/flags.go:57]:
```suggestion
		EnvVar: "WOODPECKER_HOSTNAME",
```

File: cmd/server/flags.go [hunk 5]
```diff
@@ -169,67 +169,67 @@ var flags = []cli.Flag{
...
 		Name:   "github-server",
 		Usage:  "github server address",
 		Value:  "https://github.com",
 	},
 	cli.StringFlag{
-		EnvVar: "DRONE_GITHUB_CONTEXT,WOODPECKER_GITHUB_CONTEXT",
+		EnvVar: "WOODPECKER_GITHUB_CONTEXT",
 		Name:   "github-context",
 		Usage:  "github status context",
 		Value:  "continuous-integration/drone",
```
Comment (anbraten) [inline cmd/server/flags.go:219]:
```suggestion
		Value:  "continuous-integration/woodpecker",
```

File: cmd/cli/main.go [hunk 6]
```diff
@@ -28,18 +28,18 @@ func main() {
 	app.EnableBashCompletion = true
 	app.Flags = []cli.Flag{
 		cli.StringFlag{
-			EnvVar: "DRONE_TOKEN,WOODPECKER_TOKEN",
+			EnvVar: "WOODPECKER_TOKEN",
 			Name:   "t, token",
```
Comment (anbraten) [inline cmd/cli/main.go:32]: Is this just a strange name or a way of writing alternatives like `EnvVar`?

File: cmd/cli/main.go [hunk 7]
```diff
@@ -28,18 +28,18 @@ func main() {
...
 			Usage:  "server auth token",
 		},
 
 		cli.StringFlag{
-			EnvVar: "DRONE_SERVER,WOODPECKER_SERVER",
+			EnvVar: "WOODPECKER_SERVER",
 			Name:   "s, server",
```
Comment (anbraten) [inline cmd/cli/main.go:38]: Strange name as well.

File: cmd/server/flags.go [hunk 8]
```diff
@@ -240,232 +240,232 @@ var flags = []cli.Flag{
...
+		EnvVar: "WOODPECKER_GITEA",
 		Name:   "gitea",
 		Usage:  "gitea driver is enabled",
 	},
 	cli.StringFlag{
-		EnvVar: "DRONE_GITEA_URL,WOODPECKER_GITEA_URL",
+		EnvVar: "WOODPECKER_GITEA_URL",
 		Name:   "gitea-server",
 		Usage:  "gitea server address",
 		Value:  "https://try.gitea.io",
```
Comment (anbraten) [inline cmd/server/flags.go:307]:
```suggestion
```

We should show an error if the user wants Gitea and forgot the url.

File: cmd/cli/main.go (hunk 6 above)
Comment (6543) [reply cmd/cli/main.go:32]: I would add a TODO but not start refactoring this here

File: cmd/server/flags.go (hunk 8 above)
Comment (6543) [reply cmd/server/flags.go:307]: agree!

File: cmd/cli/main.go (hunk 6 above)
Comment (anbraten) [reply cmd/cli/main.go:32]:
```suggestion
                        // TODO: rename to `token`
			Name:   "t, token",
```

File: cmd/cli/main.go (hunk 7 above)
Comment (anbraten) [reply cmd/cli/main.go:38]:
```suggestion
			// TODO: rename to `server`
			Name:   "s, server",
```

**Review by 6543 — APPROVED** [review]

**Review by anbraten — APPROVED** [review]

**Comment by marians** [comment]
Thanks for taking over!

**Pull request merged**